src/account.c

```diff
@@ -1719,39 +1719,81 @@ const char *account_extra(const struct account *acc)
  */
 int account_uri_complete(const struct account *acc, struct mbuf *buf,
 			 const char *uri)
+{
+	char *str;
+	struct pl pl;
+	int err;
+
+	pl_set_str(&pl, uri);
+	err = account_uri_complete_strdup(acc, &str, &pl);
+	if (err)
+		return err;
+
+	err = mbuf_write_str(buf, str);
+	mem_deref(str);
+
+	return err;
+}
+
+
+/**
+ * Auto complete a SIP uri, add scheme and domain if missing
+ *
+ * @param acc  User-Agent account
+ * @param strp Pointer to destination string; allocated and set
+ * @param uri  Input SIP uri pointer-length string
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int account_uri_complete_strdup(const struct account *acc, char **strp,
+				const struct pl *uri)
 {
 	struct sa sa_addr;
-	size_t len;
 	bool uri_is_ip;
 	char *uridup;
 	char *host;
 	char *c;
+	bool append = true;
+	struct mbuf *mb;
+	struct pl trimmed;
+	bool complete = false;
 	int err = 0;
 
-	if (!buf || !uri)
+	if (!strp || !pl_isset(uri))
 		return EINVAL;
 
 	/* Skip initial whitespace */
-	while (isspace(*uri))
-		++uri;
+	trimmed = *uri;
+	err = pl_ltrim(&trimmed);
+	if (err)
+		return err;
```

Comment:
this seems to mix two types of changes; refactoring and logical changes.

even if more tedious to do, it is better to keep those changes in separate PR's